operator/Makefile

```diff
@@ -1,20 +1,21 @@
 # GOBUILD relies on the order of makefile list to get VERSION file
 include ../Makefile.defs
 
-TARGET=cilium-operator
-SOURCES := $(shell find ../pkg . \( -name '*.go'  ! -name '*_test.go' \))
-$(TARGET): $(SOURCES)
-	@$(ECHO_GO)
-	$(QUIET)$(GO) build $(GOBUILD) -o $(TARGET)
+TARGETS := cilium-operator cilium-operator-none cilium-operator-aws cilium-operator-azure
```

Comment:
I'm not too keen on `cilium-operator-none`, perhaps `cilium-operator-generic` or `cilium-operator-minimal`  would be better?

Part of me also things we could name these more verbosely, i.e. `cilium-operator-ipam-aws` and `cilium-operator-ipam-azure` or even `cilium-operator-with-aws-ipam`, but shorter names seem just fine actually, except for one that is provider-free...